### PR TITLE
Add Missing Entity API: Cat

### PR DESCRIPTION
**Pull request by Owen1212055**
Cat Resting
image

Cat Head Up
image

Requires: https://github.com/PaperMC/Paper/pull/5743

### CODE REVIEW

**Comment by Machine-Maker** [comment]
I am of the opinion that these more mob behaviors should be in a single patch, cause I'm sure there are others missing. I think the wolf is missing a beg one at least. Then if spigot adds them themselves, they are all in one place to remove the conflicting ones.

**Comment by Owen1212055** [comment]
> I am of the opinion that these more mob behaviors should be in a single patch, cause I'm sure there are others missing. I think the wolf is missing a beg one at least. Then if spigot adds them themselves, they are all in one place to remove the conflicting ones.

So just combine this one into the fox API patch?

**Comment by Machine-Maker** [comment]
They can probably be two PRs, but wait until one is merged and then do the other one, or rebase this one off of your other branch so it’s clear this requires the other to be merged first, or you can combine them into one pr. Plenty of options.

**Comment by Owen1212055** [comment]
Updated to 1.17

**Comment by Owen1212055** [comment]
rebased

File: patches/api/0320-Missing-Entity-Behavior-API.patch
Comment (Machine-Maker) [inline patches/api/0320-Missing-Entity-Behavior-API.patch]: idk, I've never liked the setIsResting pattern. Doesn't setResting mean the same, and sound way better? Or it looks like the NMS method is called setLying, that works too. No reason it can't be the same I don't think.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by Owen1212055** [comment]
Renamed

**Review by Machine-Maker — APPROVED** [review]

**Review by jpenilla — CHANGES_REQUESTED** [review]
Whether the cat is lying down or not doesn't effect behavior (i.e. a cat set to lie down will slide around the floor when it wants to walk). This is because in vanilla the cat is set to lie down/get up by the `CatLieOnBedGoal` and `Cat.CatRelaxOnOwnerGoal` goals.

I think the following things should be included in the documentation:
- Lying down does not effect behavior
- That this API is probably best used in combination with the mob goals API

Besides that, this looks fine.

**Comment by Owen1212055** [comment]
> Whether the cat is lying down or not doesn't effect behavior (i.e. a cat set to lie down will slide around the floor when it wants to walk). This is because in vanilla the cat is set to lie down/get up by the `CatLieOnBedGoal` and `Cat.CatRelaxOnOwnerGoal` goals.
> 
> 
> 
> I think the following things should be included in the documentation:
> 
> - Lying down does not effect behavior
> 
> - That this API is probably best used in combination with the mob goals API
> 
> 
> 
> Besides that, this looks fine.

Maybe this patch should be renamed then. Generally this patch contains only visual changes. See: all the horse api

I could rename the patch to something like entity appearance.

**Comment by Owen1212055** [comment]
> Whether the cat is lying down or not doesn't effect behavior (i.e. a cat set to lie down will slide around the floor when it wants to walk). This is because in vanilla the cat is set to lie down/get up by the `CatLieOnBedGoal` and `Cat.CatRelaxOnOwnerGoal` goals.
> 
> I think the following things should be included in the documentation:
> 
> * Lying down does not effect behavior
> * That this API is probably best used in combination with the mob goals API
> 
> Besides that, this looks fine.

Additionally, I want to add onto this patch that will most likely include more changes like these.  This patch already has changes that are just like the cat things, see: Fox API. Unless we want to rename the patch to something else. I currently have a couple pull requests that are waiting on this pull request, so if we can get this resolved as soon as possible that would be great.

**Comment by Owen1212055** [comment]
Rebased and updated. I added a little bit to the documentation.

**Review by kennytv — APPROVED** [review]

**Review by jpenilla — APPROVED** [review]

**Pull request merged**
